tests/test_config_entries.py

```diff
@@ -2611,29 +2619,48 @@ async def test_entry_setup_invalid_state(
     assert entry.state is state
 
 
-async def test_entry_unload_succeed(
-    hass: HomeAssistant, manager: config_entries.ConfigEntries
+@pytest.mark.parametrize(
+    ("unload_result", "expected_result", "expected_state", "has_runtime_data"),
+    [
+        (True, True, config_entries.ConfigEntryState.NOT_LOADED, False),
+        (False, False, config_entries.ConfigEntryState.LOADED, True),
```

Comment:
It is also weird that the config entry is still in state `ConfigEntryState.LOADED` when unload fails. In a follow-up, I think the implementation should be adjusted to change states to `ConfigEntryState.FAILED_UNLOAD` when this happens because the state of the config entry is most likely in some half broken state.